archive/k/kotlin/reverse-string.kt

```diff
@@ -0,0 +1,12 @@
+fun String.reverseCaseOfString(): String {
+    val inputCharArr = toCharArray() // Converting the input to char array
+    var output = ""
+    for (i in 0 until inputCharArr.size) {
+        output += if (inputCharArr[i].isUpperCase()) { // Checking the character is in uppercase or not
+            inputCharArr[i].toLowerCase() // Converting the char to lower case
+        } else {
+            inputCharArr[i].toUpperCase() // Converting the char to upper case
+        }
+    }
+    return output
+}
```

Comment:
Also, this ought to execute on its own. I'm not familiar with Kotlin, but if it's related to Java at all then you can make a call to this method in the main method. Just make sure to pull the string to be reversed off the command line.